streams/src/main/java/org/apache/kafka/streams/state/internals/CachingKeyValueStore.java

```diff
@@ -155,6 +169,11 @@ private void putInternal(final Bytes key,
                 context.timestamp(),
                 context.partition(),
                 context.topic()));
+        if (context.recordMetadata().isPresent() && consistencyEnabled) {
```

Comment:
We need to record the position here as well as reads might be served directly from the cache